redis/pubsub_test.go

```diff
@@ -74,3 +76,25 @@ func TestPushed(t *testing.T) {
 		t.Errorf("recv /w timeout got %v, want %v", got, want)
 	}
 }
+
+func TestPubSubReceiveContext(t *testing.T) {
+	sc, err := redis.DialDefaultServer()
+	if err != nil {
+		t.Fatalf("error connection to database, %v", err)
+	}
+	defer sc.Close()
+
+	c := redis.PubSubConn{Conn: sc}
+
+	require.NoError(t, c.Subscribe("c1"))
+	expectPushed(t, c, "Subscribe(c1)", redis.Subscription{Kind: "subscribe", Channel: "c1", Count: 1})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	got := c.ReceiveContext(ctx)
+	if err := got.(error); !errors.Is(err, context.Canceled) {
```

Comment:
If this doesn't return an error it will panic, which I don't believe is the intention?

Reply:
Added checks for this.